src/counter/handler.cpp

```diff
@@ -51,9 +51,12 @@ void handler::on_master_start() {
             make_server_socket(nullptr, port, w),
             cybozu::reactor::EVENT_IN);
     } else {
-        m_reactor.add_resource(
-            make_server_socket(g_config.vip(), port, w, true),
-            cybozu::reactor::EVENT_IN);
+        if( g_config.vip() ) {
```

Comment:
Oh, that's right.

Reply:
Since `m_vip` is always set, I think using `optional` for this
might be confusing and lead to bugs.

I'll leave this to you on how to fix it.

Reply:
When `leader_election_method` is `file`, yrmcds actually has no virtual IPs. Therefore, I feel that the type of `m_vip` should be `std::optional` to model the situation correctly.

What is wrong is the initial value of `m_vip`. It should be initialized to be `nullopt` and should only have the user-specified value or `127.0.0.1` when in `virtual_ip` mode.

Reply:
↑This was wrong.

Since the initial value of `leader_election_method` is `virtual_ip`, the initial value of `m_vip` must be `127.0.0.1`. 
Therefore, we must empty `m_vip` when `leader_election_method` is changed to other than `virtual_ip`.